core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -144,6 +144,35 @@ class LogSegmentTest {
     checkEquals(ms2.records.iterator, read.records.records.iterator)
   }
 
+  @Test
+  def testReadWhenNoMaxPosition(): Unit = {
+    val maxPosition: Optional[java.lang.Long] = Optional.empty()
+    val maxSize = 1
+    val seg = createSegment(40)
+    val ms = records(50, "hello", "there")
+    seg.append(51, RecordBatch.NO_TIMESTAMP, -1L, ms)
+    for (minOneMessage <- Array(true, false)) {
```

Comment:
Could we use `@ParameterizedTest` to test it? Otherwise, if the test fail, it's not clear which case causing the failure.
```
  @ParameterizedTest
  @ValueSource(booleans = Array(true, false))
```